### PR TITLE
#82: Remove setup.py and update poetry to 1.4.0

### LINKED ISSUE #82: Remove setup.py
# Background

- with poetry 1.4.0 it doesn't create the setup.py anymore https://github.com/python-poetry/poetry/releases/tag/1.4.0
- we currently use poetry build to generate the setup.py
- however, the setup.py isn't needed with newer pip versions and if there are releases to pypi or as wheels
- for that reason, we can remove setup.py and githook that generates it, from this repo

# Acceptance Criteria

- [ ] Update workflows to poetry 1.4.0
- [ ] Remove setup.py Github Workflow
- [ ] Remove setup.py githook
- [ ] Remove setup.py file

### PR DESCRIPTION
**Pull request by tkilias**
Fixes #82

### CODE REVIEW

**Comment by tkilias** [comment]
We decided to fix the [AWS CI issues](https://github.com/exasol/sagemaker-extension/issues/85) later and rely for the moment on local tests.

File: pyproject.toml [hunk 1]
```diff
@@ -22,7 +22,8 @@ include = ["lua", "resources", "deployment"]
 [tool.poetry.dependencies]
 python = ">=3.8,<4.0"
 pandas = ">=1.4.2,<2.0.0"
-boto = "^2.49.0"
+boto3 = "1.26.163"
+botocore = "1.29.163"
```
Comment (kaklakariada) [inline pyproject.toml:26]: Downgrade of boto by intention?

**Review by kaklakariada — APPROVED** [review]

File: scripts/setup_integration_test.sh [hunk 2]
```diff
@@ -20,7 +20,7 @@ function install_localstack {
 function checkout_exasol_test_container {
   cnt_func=$((cnt_func+1))
   echo -e "${YEL} Step-$cnt_func: ${GRA} Checkout Exasol test container ${NC}"
-   git clone https://github.com/exasol/integration-test-docker-environment.git
+   git clone --branch 1.6.0 https://github.com/exasol/integration-test-docker-environment.git
```
Comment (MarleneKress79789) [inline scripts/setup_integration_test.sh:23]: why pin the version?

**Review by MarleneKress79789 — APPROVED** [review]
checked everything except the java file. looks fine to me

**Pull request merged**